bindings/rust/bench/src/harness.rs

```diff
@@ -485,14 +486,17 @@ mod tests {
         }
     }
 
-    fn session_resumption<C: TlsConnection, S: TlsConnection>() {
+    fn session_resumption<C, S>()
```

Comment:
I'm not sure if you can fix it in this PR, but sooner or later it's going to get really confusing to keep everything in the bench folder. Just somehow adding separation between benchmarking stuff and generic TLS stuff would be helpful. Because right now some of these tests are interop tests, but they're in the benchmarking folder, which is a really weird place to end up.

Reply:
I agree, but I specifically didn't want to fix it in this PR since _technically_ I'm not changing any functionality, and it's already a little funky to read. 

My thought it that is probably makes sense to move the TlsConnection implementation to the integration crate, and then just have the TlsBenchConfig trait in the bench crate. I can do that in the next PR.